src/core/qgsvectorlayer.cpp

```diff
@@ -2422,6 +2422,20 @@ bool QgsVectorLayer::readStyle( const QDomNode &node, QString &errorMessage,
         QDomElement e = layerOpacityNode.toElement();
         setOpacity( e.text().toDouble() );
       }
+
+      const bool hasScaleBasedVisibiliy { node.attributes().namedItem( QStringLiteral( "hasScaleBasedVisibilityFlag" ) ).nodeValue() == '1' };
+      setScaleBasedVisibility( hasScaleBasedVisibiliy );
+      bool ok;
+      const double maxScale { node.attributes().namedItem( QStringLiteral( "maxScale" ) ).nodeValue().toDouble( &ok ) };
```

Comment:
maybe safer to check returned namedItem isNull node instead delegate a side effect of nodeValue return to empty string?

Reply:
No need to. Converting an empty string to double sets ok to false.

Reply:
I know that! but what else if the api change and nodeValue return None (ore something else) instead of string? Because node isNull it's not garanteed that nodeValue should be empty string in the future.